### PR TITLE
[Docs] Clarify HTTPS Termination Proxy on Using Reverse Proxy to Enable HTTPS

### LINKED ISSUE #9354: [Docs] Clarify TLS Termination Proxy
# Description

On [HTTPS Setup](https://docs.gitea.io/en-us/https-setup/), reverse proxy can be used for providing HTTPS instance. In reality, such setup is called [TLS Termination Proxy](https://en.wikipedia.org/wiki/TLS_termination_proxy).

Perhaps the docs can be rewritten as:

```
You can also enable HTTPS by using reverse proxy. Set up your favorite reverse proxy (Apache,
NGINX, or Caddy) and follow documentation of the program of your choice to enable HTTPS site.

NOTE: Enabling HTTPS by this method is referred as TLS Termination Proxy. The reverse proxy
handle TLS connections, and decrypted them for passing to Gitea. This is fine as long as the reverse
proxy and Gitea instance are either on the same machine, or on different machine on private
network (the proxy is exposed to outside network). If your Gitea instance is on public network, and
you want to enable HTTPS with reverse proxy, it is recommended to also encrypt connections from
your proxy to Gitea by follow instructions for built-in server and change proxy URL to https://[URL].
```

### PR DESCRIPTION
**Pull request by bagasme**
Indeed, enabling HTTPS by reverse proxy often called *TLS Termination Proxy*. It is optional to also combine it with enabling HTTPS with internal server in case of public instances.

This will close #9354.

### CODE REVIEW

**Comment by bagasme** [comment]
Codecov report seems weird, as only updating docs can decrease coverage only slightly on supposedly random files?

**Review by lafriks — APPROVED** [review]

File: docs/content/doc/usage/https-support.md [hunk 1]
```diff
@@ -76,5 +76,6 @@ After that, enable HTTPS by following one of these guides:
 * [apache2/httpd](https://httpd.apache.org/docs/2.4/ssl/ssl_howto.html)
 * [caddy](https://caddyserver.com/docs/tls)
 
-Note: Your connection between your reverse proxy and Gitea might be unencrypted. To encrypt it too, follow the [built-in server guide](#using-built-in-server) and change
-the proxy url to `https://[URL]`.
+Note: Enabling HTTPS by this method is referred as [TLS Termination Proxy](https://en.wikipedia.org/wiki/TLS_termination_proxy). The reverse proxy handles TLS connections, and decrypted them for passing to Gitea. This is fine as long as both the proxy and Gitea instances are either on the same machine, or on different machines at private network (with the proxy is exposed to outside network. If your Gitea instance is at public network instead, you might also consider to [enable HTTPS using built-in server](#using-the-built-in-server) to encrypt connections from the proxy to Gitea.
```
Comment (mrsdizzie) [inline docs/content/doc/usage/https-support.md]:
```suggestion
Note: Enabling HTTPS only at the proxy level is known as a [TLS Termination Proxy](https://en.wikipedia.org/wiki/TLS_termination_proxy). The proxy server accepts incoming TLS connections, decrypts the contents, and passes the now unencrypted contents to the Gitea application. This is normally fine as long as both the proxy and Gitea instances are either on the same machine, or on different machines within the same private network. If your Gitea instance is separated from your Proxy over a public network, or you want full end-to-end encryption, you can also enable HTTPS support directly in Gitea [using the built-in server](#using-the-built-in-server) and forward the connections over HTTPS instead.
```
Some small adjustments for English/clarity

**Review by mrsdizzie — CHANGES_REQUESTED** [review]

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
@bagasme Please note @mrsdizzie's comment.

File: docs/content/doc/usage/https-support.md (hunk 1 above)
Comment (bagasme) [reply docs/content/doc/usage/https-support.md]: OK

**Comment by lunny** [comment]
make L-G-T-M work

**Comment by mrsdizzie** [comment]
It looks like my suggestion wasn't applied from Github but just some of it was retyped. There are still some typos and missing words.

**Comment by bagasme** [comment]
@mrsdizzie Feel free to open new PR with your suggestions.

**Pull request merged**